### DISCUSSION TITLE
Layout jumps occur if a client component wraps around a server component

### DISCUSSION THREAD

**Discussion by dmproj** [vercel/next.js#71894] [App Router]
hi, I'm experiencing layout jumps, and it also draws some additional lines when the client component wraps around the server component. Everything works fine without the wrapper. I also have another two components that have no lines but still experience jumps
![New Bitmap Image (2)](https://github.com/user-attachments/assets/2c61732c-af4e-4e08-ba1a-45d3de71e77a)
its not the css issue, appreciate if you could help with the matter

```
"use client";
import { usePathname } from "next/navigation";
import Link from "next/link";
import styles from "./ActiveLink.module.css";

const ActiveLink = ({ href, children }) => {
  const pathname = usePathname();
  const isActive = pathname === href;

  return (
    <Link href={href} className={`${styles.navlinkStyles} ${isActive ? styles.active : ""}`}>
      {children}
    </Link>
  );
};

export default ActiveLink;
```

```
import styles from  "./JobsMainLeft.module.css"
import ActiveLink from "./ActiveLink";
import Image from "next/image";

export default function JobsMainRight({ jobs }) {
  return (
    <div>
      <div>Total Jobs: {jobs.length}</div>
      {jobs.map((job) => (
        <ActiveLink 
          key={job._id} 
          href={`/jobs/${job.fleet}/${encodeURIComponent(job.titleUrl.replace(/\s+/g, '-').toLowerCase())}`}
        >
          <div className={styles.jobCard}>
            <div className={styles.logoContainer}>
              {job.company.logo.length > 0 ? (
                <Image src={job.company.logo[0].url} alt="Company Logo" className={styles.logo} />
              ) : (
                <div className={styles.noLogo}>No Logo</div>
              )}
            </div>
            <div className={styles.jobDetails}>
              <div className={styles.companyName}>
                {job.company.companyName} - {job.urgency}
              </div>
              <div className={styles.jobTitle}>{job.title}</div>
              <div className={styles.salary}>
                {job.salary} {job.salaryType}
              </div>
            </div>
          </div>
        </ActiveLink>
      ))}
    </div>
  );
}

```

**Comment by aniruddhaadak80** [comment]
Hi @dmproj ,

Layout jumps can occur when a client component wraps around a server component due to differences in rendering behavior between the two. In your case, the `ActiveLink` component is a client component that interacts with `usePathname`, which may introduce re-renders or changes in layout during navigation.

Here are a few suggestions to troubleshoot and potentially resolve the issue:

1. **Check the Wrapper Structure**: Ensure that the structure of your wrapper and the server component is optimal. Sometimes adjusting how components are nested can mitigate layout shifts.

2. **CSS Styles**: Since you mentioned it’s not a CSS issue, double-check if the styles of the client component may inadvertently cause layout changes during re-renders. Pay particular attention to margin, padding, and positioning properties.

3. **Using Layouts**: If possible, try using a layout component that encapsulates both client and server components. This could help maintain a consistent layout during state changes.

4. **Avoid Conditional Rendering**: If you're conditionally rendering elements based on state or props, it could lead to layout shifts. Ensure that elements are consistently rendered, even if they're hidden.

5. **Image Handling**: Since you're using `next/image`, ensure that the `layout` prop is properly set (e.g., `fill`, `fixed`, `intrinsic`) to maintain the aspect ratio and prevent jumps.

6. **Debugging Tools**: Utilize browser dev tools to monitor layout shifts and identify specific elements causing the jumps.

If the issue persists, providing additional details or specific scenarios where the jumps occur can help in diagnosing the problem further.

Thanks!

**Comment by icyJoseph** [comment]
Any link where we can see the problem live? I reckon this is some CSS/HTML issue 🤔
